protocol/proto/openyolo.proto

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+syntax = "proto3";
+
+package org.openyolo;
+
+option java_package = "org.openyolo.protocol";
+option java_outer_classname = "Protobufs";
+
+message CredentialRetrieveRequest {
+    repeated string       authMethods        = 1; // at least one required
+    map<string, bytes> additionalProps      = 2;
+}
+
+message CredentialRetrieveResponse {
+    bytes              retrieveIntent   = 1; // required
+    map<string, bytes> additionalProps = 2;
+}
+
+message CredentialRetrieveResponseList {
+    repeated CredentialRetrieveResponse responses = 1;
+}
+
+message HintRetrieveRequest {
+    repeated string       authMethods      = 1; // at least one required
+    repeated string       idTypes          = 2; // at least one required
+    map<string, bytes>    additionalProps = 3;
+    PasswordSpecification passwordSpec     = 4;
+}
+
+message HintRetrieveResponse {
+    bytes              retrieveIntent = 1; // required
+    map<string, bytes> additionalProps = 2;
+}
+
+message HintRetrieveResponseList {
+    repeated HintRetrieveResponse responses = 1;
+}
+
+message Credential {
+    string id                    = 1; // required
+    string authDomain            = 2; // required
+    string authMethod            = 3; // required
+    string displayName           = 4;
+    string displayPictureUri     = 5;
+    string password              = 6;
+    map<string, bytes> additionalProps = 7;
```

Comment:
There are a couple proto fields that do follow the equals sign horizontal alignment in this file.  We should standardize on our proto format.  Personally my vote is to not enforce horizontal alignment.